commands/uploader.go

```diff
@@ -289,6 +289,8 @@ func (c *uploadContext) ReportErrors() {
 		}
 
 		if !c.allowMissing {
+			Print("hint: Your push was rejected due to missing or corrupt local objects.\n" +
```

Comment:
I like this message, and I think that it's good that we're only showing it once at the end of the attempted push, right before calling os.Exit.

If you don't mind, I think that we could extract this message out to a `var` and print it by referencing that instead of having a string constant in the code. We use this pattern in the `clone` and `checkout` commands, which have deprecation notices.

I was thinking something like:

```go
var (
        pushMissingHint = []string{...}
)

if !c.allowMissing {
        Print(strings.Join(pushMissingHint, "\n"))
}
os.Exit(2)
```